tools/underwater-gps.py

```diff
@@ -79,101 +190,82 @@ def format(message, now=0, lat=0, lon=0, orientation=0):
-    )
-    master.mav.gps_raw_int_send(
-        0,                     # time_usec                 : Timestamp (microseconds since UNIX epoch or microseconds since system boot) (uint64_t)
-        3,                     # fix_type                  : See the GPS_FIX_TYPE enum. (uint8_t)
-        result['lat'] * 1e7,   # lat                       : Latitude (WGS84), in degrees * 1E7 (int32_t)
-        result['lon'] * 1e7,   # lon                       : Longitude (WGS84), in degrees * 1E7 (int32_t)
-        0,                     # alt                       : Altitude (AMSL, NOT WGS84), in meters * 1000 (positive for up). Note that virtually all GPS modules provide the AMSL altitude in addition to the WGS84 altitude. (int32_t)
-        0,                     # eph                       : GPS HDOP horizontal dilution of position (unitless). If unknown, set to: UINT16_MAX (uint16_t)
-        0,                     # epv                       : GPS VDOP vertical dilution of position (unitless). If unknown, set to: UINT16_MAX (uint16_t)
-        0,                     # vel                       : GPS ground speed (m/s * 100). If unknown, set to: UINT16_MAX (uint16_t)
-        0,                     # cog                       : Course over ground (NOT heading, but direction of movement) in degrees * 100, 0.0..359.99 degrees. If unknown, set to: UINT16_MAX (uint16_t)
-        6                      # satellites_visible        : Number of satellites visible. If unknown, set to 255 (uint8_t)
-    )
-    master.mav.vfr_hud_send(
-        0,                     # airspeed                  : Current airspeed in m/s (float)
-        0,                     # groundspeed               : Current ground speed in m/s (float)
-        result['orientation'], # heading                   : Current heading in degrees, in compass units (0..360, 0=north) (int16_t)
-        0,                     # throttle                  : Current throttle setting in integer percent, 0 to 100 (uint16_t)
-        0,                     # alt                       : Current altitude (MSL), in meters (float)
-        0                      # climb                     : Current climb rate in meters/second (float)
-    )
-
     # new approach: nmea messages to port 14401
     try:
-        result = response.json()
         for message in [gpgga, gprmc, gpvtg]:
-            msg = format(message, datetime.now(), result['lat'], result['lon'], orientation=result['orientation'])
+            msg = format(
+                message,
+                datetime.now(),
+                result['lat'],
+                result['lon'],
+                orientation=result['orientation']
+                )
             qgcNmeaSocket.sendto(msg, ('192.168.2.1', 14401))
-    except Exception as e:
-        print(e)
+    except Exception as error:
+        report_status("Error reading master position: {0}".format(error))
```

Comment:
In some points you are doing {0}.format(error) and in the others str(error), please be consistent